core/chains/evm/client/rpc_client.go

```diff
@@ -122,7 +123,7 @@ type RPCClient struct {
 }
 
 var _ commonclient.RPCClient[*big.Int, *evmtypes.Head] = (*RPCClient)(nil)
-var _ commonclient.SendTxRPCClient[*types.Transaction] = (*RPCClient)(nil)
+var _ commonclient.SendTxRPCClient[*types.Transaction, *SendTxResult] = (*RPCClient)(nil)
 
 func NewRPCClient(
 	cfg config.NodePool,
```

Comment:
we already pass NodePool config. We can avoid passing clientError as a separate param